src/lib/canvas/canvas.js

```diff
@@ -19,20 +20,33 @@ const Container = styled.div`
 class Canvas extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      creatorListId: props.creatorListId,
+      creators: props.creators,
+      layout: {
+        id: guid(),
+        type: 'element'
+      }
+    };
+
     this.handleUpdate = this.handleUpdate.bind(this);
-    this.state = { layout: this.props.layout };
   }
 
   handleUpdate(layout) {
     this.setState({ layout });
   }
 
-  render() {
-    const { layout } = this.state;
+  componentDidMount() {
+    for (const el of document.getElementById(this.state.creatorListId).children) {
```

Comment:
can you set one event listener on a parent and then identify on el dragstart has appered?
Also I don't see that you remove event listener on componentWillUnmount